### DISCUSSION TITLE
Failed to use “build.py” to build. Lack of “inline” module?

### DISCUSSION THREAD

**Discussion by WeiYea** [rustdesk/rustdesk#2241] [General]
error[E0583]: file not found for module `inline`
  --> src\ui.rs:27:1
   |
27 | pub mod inline;
   | ^^^^^^^^^^^^^^^
   |
   = help: to create the module `inline`, create file "src\ui\inline.rs" or "src\ui\inline\mod.rs"

**Comment by djex** [comment]
inline.rs is generated by `res/inline-sciter.py`. Are you building using the provided build.py? If so, it should be running `python3 res/inline-sciter.py` however for my setup I had to change python3 to python ->` python res/inline-sciter.py`

If your not using build.py I recommend you do. It still can be built without it however. To do so you will need to run inline-sciter.py first to generate the inline.rs source then build using cargo.

```
cargo build --release --features inline
```

**Reply by WeiYea** [reply]
Thank you. It's ready to compile. What's the function of "inline"

**Reply by djex** [reply]
inline includes the sciter html and css into the compiled exe as well as some resource files from the res folder. Without the inline feature you would need to provide the `/src/ui `folder along with the compiled exe.
